### PR TITLE
Sudoku test

### PR DESCRIPTION
**Pull request by burke-md**
## Goal:
Clarify testing procedure

## Changes:
Mostly semantic informational changes
-  Function signature in README better depicts requirements of users
- Sample code for validating 2D array (matrix) removed from `main.nr` and added to README
-  Remove Nargo tests (this puzzle will focus on Foundry testing)
-  Cleared generated smart contract from repo
-  Reset input/output values in circuit files

## Future development:
-  Actual implementation of tests
-  Script to update circuit input file between tests

### CODE REVIEW

**Comment by supernovahs** [comment]
I don't think we should remove the code and add to README .
It is not a sample code. Its part of the main puzzle which checks that the question inputs are correct

**Comment by burke-md** [comment]
@supernovahs Okay, perhaps you could help me understand the goals of this puzzle. As the question given in every sudoku that I have ever done has been valid, I assumed that it was a template of sorts or recomendation to help the users see how they might validate the answer. 

Additionally I would appreicate some insight on the testing. 

Given that Foundry is reading the proof and making a comparison, was it your intention to return a bool value? 

In previous puzzles using an assert to fail a test seemed acceptable, however we need the circuit to compile

**Comment by supernovahs** [comment]
So , the goal is to create a verifier , such that , it should verify that the solution to the question is correct, i.e it follows all the sudoku rules, like , solution should be in range of 1-4  and every number should be unique in each row and column. 
If any of the rule is not followed , it will revert due to constrain statement in the circuit . 
Due to this , I did not enter a bool return value, as if it is correct , it will pass, else will just throw error .

File: circuits/Sudoku/circuits/src/main.nr [hunk 1]
```diff
@@ -1,84 +1,18 @@
...
+//    "solution" is a 16 length array. Example: [1,4,3,2,3,2,1,4,4,1,2,3,2,3,4,1] == [1, 4, 3, 2]
+//                                                                                   [3, 2, 1, 4]
+//                                                                                   [4, 1, 2, 3]
+//                                                                                   [2, 3, 4, 1]
 
+fn main(question: [Field; 16], answer : [Field; 16]) {
     // Solution Starts Here ..
-}
-
-#[test]
```
Comment (supernovahs) [inline circuits/Sudoku/circuits/src/main.nr:77]: This can be deleted

**Comment by supernovahs** [comment]
The main problem is how to test more accurately if the solution is wrong. The purpose of the challenge for the user is to create the verification code themselves. How do we ensure they have done it correct?
If we do the checking ourselves, then there is no point in the puzzle.
Any recommendation @burke-md  as to how we should check more thoroughly?

**Comment by burke-md** [comment]
I also thought the purpose was to verify the answer, not the question. I placed the question verification code in the README as it is a great resource for users attempting to solve the puzzle. This will streamline the actual puzzle code down to the absolute minimum and I believe, encourage more users to tackle the problem.

In regards to testing, using the asserts(previously constraints), It seems to limit what you may test with Foundry. Either the circuit is proven or it fails. There is no further test to run. Additionally as puzzle creators we are counting on users to utilize the assert statement.

I'm still wrapping my head around what the right or most idiomatic way to do this is.

A potential road forward could be supplying multiple answers and expecting the circuit to prove/fail under the appropriate conditions. This could be done with either copy/paste and observe whether the circuit proves or a test that is reading the main functions output via Foundry(a bool value).

**Pull request closed without merging**